src/SpotifyLogin.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from "react"
+import { generateAuthUrl } from "./utils/spotify"
+import axios from "axios";
+
+export const SpotifyLogin = () => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [username, setUsername] = useState("");
+    const [error, setError] = useState(false);
+
+    useEffect(() => {
+        const token = sessionStorage.getItem("token");
+
+        if (!token)
+            return;
+
+        setIsLoggedIn(true);
+
+        axios
+            .get("https://api.spotify.com/v1/me", {
+                headers: {
+                    "Authorization": `Bearer ${token}`
+                }
+            })
+            .then(res => setUsername(res.data["display_name"]))
+            .catch(err => setError(true));
```

Comment:
Spotify API has good user friendly error messages - it would be good to show it to the user

Reply:
I have extracted spotify api call in separeate function, but there is a problems with error handling. When server returns 401 error then fetch throws an exception and I cannot read the message that describes the error from response. At this moment I used some workaround, such as `fetch(...).catch(() => {})`, but this is defenetly a bad approach.

Reply:
401 should not go into catch stmt. so you can handle it like
```
try {
   const response =.await fetch(...)
   if (response.status >= 400) return await parseError(response)
   return await response.json();
} catch (e) {
  // network error
}
```